scripts/postlink/ios/postlink.js

```diff
@@ -4,8 +4,20 @@ var inquirer = require('inquirer');
 var path = require("path");
 var plist = require("plist");
 
+// Fix for https://github.com/Microsoft/react-native-code-push/issues/477
+// Try to resolve package.json, assuming we are working off of the project root
+// We will use this later to fetch the correct AppDelegate and Plist file, in case there are multiple.
+var package = require(path.resolve('./package.json'));
+
 var ignoreNodeModules = { ignore: "node_modules/**" };
-var appDelegatePath = glob.sync("**/AppDelegate.m", ignoreNodeModules)[0];
+var appDelegatePaths = glob.sync("**/AppDelegate.m", ignoreNodeModules);
+
+// Fix for https://github.com/Microsoft/react-native-code-push/issues/477:
+// Typical location of AppDelegate.m for newer RN versions: $PROJECT_ROOT/ios/<project_name>/AppDelegate.m
+// Try to find that path by filtering the whole array for any path containing <project_name>
+// If we can't find it there, play dumb and pray it is the first path we find.
+var appDelegatePath = findFileByAppName(appDelegatePaths, package?package.name:null) || appDelegatePaths[0];
```

Comment:
Small nit - we generally put spaces between ternary operators (i.e. `package ? package.name : null`)